samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java

```diff
@@ -2319,6 +2371,560 @@ public String getOwner() {
+				
+				feedbackList.add(contentBuffer.toString());
+			}
+			else if (item.getData().getTypeId().equals(TypeIfc.ESSAY_QUESTION)) {
+				
+				ItemTextIfc itemText = item.getData().getItemTextArray().get(0);
+				AnswerIfc answer = itemText.getAnswerArray().get(0);
+				
+				// Key
+				key.append(ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","answer_model"));
+				key.append("\n");
+				key.append(answer.getText());
+				
+				// Feedback
+				contentBuffer.setLength(0);
+				if (item.getData().getGeneralItemFeedback() != null && !item.getData().getGeneralItemFeedback().equals(""))
+					contentBuffer.append(FormattedText.convertFormattedTextToPlaintext(item.getData().getGeneralItemFeedback()));
+								
+				feedbackList.add(contentBuffer.toString());
+			}
+			else if (item.getData().getTypeId().equals(TypeIfc.FILL_IN_BLANK) ||
+					item.getData().getTypeId().equals(TypeIfc.FILL_IN_NUMERIC)) {
+				
+				// Key
+				ItemTextIfc itemText = item.getData().getItemTextArray().get(0);
+				List<AnswerIfc> answers = itemText.getAnswerArray();
+				
+				for (int t=0; t<answers.size(); t++) {
+					AnswerIfc answer = answers.get(t);
+					if (answer.getText() == null) break;
+					
+					if (t>0) contentBuffer.append(" | ");
+					contentBuffer.append(FormattedText.convertFormattedTextToPlaintext(answer.getText()));
+				}
+				key.append(contentBuffer.toString());
+				
+				// Feedback
+				contentBuffer.setLength(0);
+				if (item.getData().getCorrectItemFeedback() != null && !item.getData().getCorrectItemFeedback().equals(""))
+					contentBuffer.append(FormattedText.convertFormattedTextToPlaintext(item.getData().getCorrectItemFeedback()));
+				
+				feedbackList.add(contentBuffer.toString());
+				
+				contentBuffer.setLength(0);
+				if (item.getData().getInCorrectItemFeedback() != null && !item.getData().getInCorrectItemFeedback().equals(""))
+					contentBuffer.append(FormattedText.convertFormattedTextToPlaintext(item.getData().getInCorrectItemFeedback()));
+				
+				feedbackList.add(contentBuffer.toString());
+			}
+  			
+  			// Responses && Feedback
+			for (int i=0; i<maxAnswers; i++) {
+				if (answerList.size() > i) {
+					row.add(answerList.get(i));
+					if (feedbackAnswerList.size() > i) row.add(feedbackAnswerList.get(i));
+					else row.add("");
+				}
+				else {
+					row.add(""); // blank answer
+					row.add(""); // blank feedback answer
+				}
+			}
+  			
+			// Key
+			row.add(key.toString());
+			
+			// Feedback correct & incorrect
+			for (int i=0; i<2; i++) {
+				if (feedbackList.size() > i)
+					row.add(feedbackList.get(i));
+				else row.add("");
+			}
+		
+			dataList.add(row);
+  		}
+  		  	  
+  		finalList.add(dataList);
+  		finalList.add(headerList);
+  		return finalList;
+    }
+
+  	private String getTypeQuestion(Long typeId) {
+  		String type = "";
+  		if (typeId.equals(TypeIfc.AUDIO_RECORDING)) {
```

Comment:
This might read better as a `switch()` statement, or at the least flip all operands in the conditionals